tests/integrations/starlette/test_starlette.py

```diff
@@ -1066,7 +1066,11 @@ def dummy_traces_sampler(sampling_context):
 
     sentry_init(
         auto_enabling_integrations=False,  # Make sure that httpx integration is not added, because it adds tracing information to the starlette test clients request.
-        integrations=[StarletteIntegration(transaction_style=transaction_style)],
+        integrations=[
+            StarletteIntegration(
+                transaction_style=transaction_style, middleware_spans=False
```

Comment:
I assume this is because the `middleware_spans` are going through `traces_sampler`. If that is the reason, we should keep the original tests and make this pass properly.

Reply:
Removed the tests change here. The sampler should be fixed by https://github.com/getsentry/sentry-python/pull/3767